client/src/main/com/sinch/sdk/domains/sms/adapters/converters/InboundsDtoConverter.java

```diff
@@ -0,0 +1,74 @@
+package com.sinch.sdk.domains.sms.adapters.converters;
+
+import com.sinch.sdk.core.exceptions.ApiException;
+import com.sinch.sdk.core.models.AbstractOpenApiSchema;
+import com.sinch.sdk.core.models.pagination.PageToken;
+import com.sinch.sdk.core.utils.Pair;
+import com.sinch.sdk.domains.sms.models.Inbound;
+import com.sinch.sdk.domains.sms.models.InboundBinary;
+import com.sinch.sdk.domains.sms.models.InboundText;
+import com.sinch.sdk.domains.sms.models.dto.v1.ApiInboundListDto;
+import com.sinch.sdk.domains.sms.models.dto.v1.InboundDto;
+import com.sinch.sdk.domains.sms.models.dto.v1.MOBinaryDto;
+import com.sinch.sdk.domains.sms.models.dto.v1.MOTextDto;
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class InboundsDtoConverter {
+
+  public static Inbound<?> convert(AbstractOpenApiSchema dto) {
+    Object obj = dto.getActualInstance();
+    if (obj instanceof MOBinaryDto) {
+      return convert((MOBinaryDto) obj);
+    } else if (obj instanceof MOTextDto) {
+      return convert((MOTextDto) obj);
+    } else {
+      throw new ApiException("Unexpected class:" + obj.getClass().getName());
+    }
+  }
+
+  public static Pair<Collection<Inbound<?>>, PageToken<Integer>> convert(ApiInboundListDto dto) {
+    // check end of pagination limit reached: (current page number * page size ) cannot be greater
+    // than "count" to be able to call next page
+    Integer nextPageToken =
+        ((dto.getPage() + 1) * Long.valueOf(dto.getPageSize())) >= dto.getCount()
```

Comment:
@JPPortier I think @Dovchik already mentioned this line on Groups PR :)

Reply:
Exact. I planned to work onto refactoring to share common logic but outside of these PRs scope because of I'll handle too the Numbers API
So cross domains refactoring I just do not wish to push with this specific SMS domain's related work